File: uxmobile/src/main/java/sk/brecka/uxmobile/model/TouchEvent.java
package sk.brecka.uxmobile.model;

import org.json.JSONArray;
import org.json.JSONException;

/**
 * Created by matej on 24.10.2017.
 */

public abstract class TouchEvent extends Event {
    private static final String TAG = "TouchEvent";

    protected static final int INDEX_X = 2;
    protected static final int INDEX_Y = 3;

    private final int mX;
    private final int mY;

    protected TouchEvent(long startTime, int x, int y) {
        super(startTime);
        mX = x;
        mY = y;
    }

    protected abstract String getType();

    public JSONArray toJson() throws JSONException {
        return super.toJson()
                .put(INDEX_X, mX)
                .put(INDEX_Y, mY);
    }
}

File: uxmobile/src/main/java/sk/uxtweak/uxmobile/model/event/TouchEvent.kt
package sk.uxtweak.uxmobile.model.event

import org.json.JSONArray

abstract class TouchEvent(startTime: Long, val x: Float, val y: Float) : Event(startTime) {
    override fun toJson(): JSONArray = super.toJson()
        .put(INDEX_X, x)
        .put(INDEX_Y, y)

    companion object {
        const val INDEX_X = 2
        const val INDEX_Y = 3
    }
}
